lib/sshutils/scp/local.go

```diff
@@ -49,7 +49,7 @@ func (l *localFileSystem) Chtimes(path string, atime, mtime time.Time) error {
 // MkDir creates a directory
 func (l *localFileSystem) MkDir(path string, mode int) error {
 	fileMode := os.FileMode(mode & int(os.ModePerm))
-	err := os.MkdirAll(path, fileMode)
+	err := os.Mkdir(path, fileMode)
```

Comment:
This differed from OpenSSH's behavior which does not automatically create intermediate directories. If that's a concern, we can revert this change.

Reply:
Adhering to OpenSSH behaviour seems more logical to me. When people type `scp` they expect it to behave just like scp does when it comes to things like this.

Reply:
I agree, we should be as consistent as possible with OpenSSH.

Reply:
@alex-kovoy Any reason you made this `MkdirAll`? https://github.com/gravitational/teleport/pull/2055

Reply:
@russjones That was original implementation, it appeared in my PR because I move some code around while working on HTTP transfers. See the original (deleted) `scp.go` file in that PR.

Reply:
Hrm, interesting the original commit seems to indicate `MkdirAll` lines up with OpenSSH behavior: https://github.com/gravitational/teleport/pull/788/commits/80df1bbc2d68f9a7aa7103ad4eb7eeac4b8b7700

Reply:
The original commit that introduced this changes is https://github.com/gravitational/teleport/commit/2ad2fb4ff1f9874a4f6265974518c42f0561ec70 and it seems to be fixing this [problem](https://github.com/gravitational/teleport/issues/274#issuecomment-198626834) from https://github.com/gravitational/teleport/issues/274.
Since the solution caused another issue, it was changed in https://github.com/gravitational/teleport/commit/c6c77a183c9a32d076dd0110eebfdc35d249ad96 but `os.MkdirAll` remained.

Reply:
As for the behavior of OpenSSH (7.4p1 from above) and the following source layout:
```
$ tree foo
foo
├── foo1
└── foo-dir
    └── foo-file
```

Copying to a non-existing directory (`zz`, 1 level), renames the source into it (or creates it remotely and copies the contents of the source directory):
```
$ scp -r ./foo centos@35.226.107.203:zz
foo1                                                                                     100%    0     0.0KB/s   00:00    
foo-file                                                                                 100%    0     0.0KB/s   00:00
```
with the resulting remote layout:
```
# tree zz
zz
├── foo1
└── foo-dir
    └── foo-file
```
while opening into a directory with an intermediate non-existing directory:
```
$ scp -r ./foo centos@35.226.107.203:non-existing/zz
scp: non-existing/zz: No such file or directory
```
but it works with tsh in `5.2.1`.